src/clients/crux-wallet-onboarding-client.ts

```diff
@@ -0,0 +1,153 @@
+import { CruxAssetTranslator } from "../core/entities/crux-asset-translator";
+import { DomainRegistrationStatus } from "../core/entities/crux-domain";
+import { CruxDomain } from "../core/entities/crux-domain";
+import { ICruxAssetTranslatorRepository, ICruxAssetTranslatorRepositoryConstructor } from "../core/interfaces/crux-asset-translator-repository";
+import { ICruxDomainRepository, ICruxDomainRepositoryConstructor } from "../core/interfaces/crux-domain-repository";
+import { IKeyManager } from "../core/interfaces/key-manager";
+import { setCacheStorage } from "../index";
+import { BasicKeyManager } from "../infrastructure/implementations/basic-key-manager";
+import { BlockstackCruxAssetTranslatorRepository } from "../infrastructure/implementations/blockstack-crux-asset-translator-repository";
+import { BlockstackCruxDomainRepository } from "../infrastructure/implementations/blockstack-crux-domain-repository";
+import { ManualCruxAssetTranslatorRepository } from "../infrastructure/implementations/manual-crux-asset-translator-repository";
+import { IClientAssetMapping, IGlobalAssetList } from "../packages/configuration-service";
+import { ErrorHelper, PackageErrorCode } from "../packages/error";
+import { InMemStorage } from "../packages/inmem-storage";
+import { getLogger } from "../packages/logger";
+import { IBlockstackServiceInputOptions } from "../packages/name-service/blockstack-service";
+import { StorageService } from "../packages/storage";
+const log = getLogger(__filename);
+// DDD Experimental Stuff
+export interface ICruxOnBoardingClientOptions {
+    cacheStorage?: StorageService;
+    configKey?: string|IKeyManager;
+    domain?: string;
+    assetMapping?: IClientAssetMapping;
+}
+export class CruxOnBoardingClient {
+    private _initPromise: Promise<void>;
+    private _cruxDomainRepository: ICruxDomainRepositoryConstructor;
+    private _cruxAssetTranslatorRepository: ICruxAssetTranslatorRepositoryConstructor;
+    private _configKeyManager?: IKeyManager;
+    private _cruxDomain?: CruxDomain;
+    private _cruxAssetTranslator?: CruxAssetTranslator;
+    private _domainContext?: string;
+    private _assetMapping?: IClientAssetMapping;
+    constructor(options: ICruxOnBoardingClientOptions) {
```

Comment:
Please simplify the client more as you're removing assetTranslator from your part of the code. 

Only `cacheStorage` makes sense here tbh. And maybe `domain`